src/Disks/tests/gtest_disk_backup.cpp

```diff
@@ -0,0 +1,105 @@
+#include <filesystem>
+#include <fstream>
+#include <memory>
```

Comment:
Also need an integration test which creates the disk with type backup and checks that we can read from it and that an attempt to write to it fails. Also check that server can be restarted (via `node.restart_clickhouse()`). Also check the same for customly created disk from SQL (mentioned in the comment above).